cmd/argocd/commands/project.go

```diff
@@ -635,24 +655,29 @@ func modifyResourceListCmd(cmdUse, cmdDesc, examples string, clientOpts *argocdc
 		Run: func(c *cobra.Command, args []string) {
 			ctx := c.Context()
 
-			if len(args) != 3 {
+			if len(args) != 3 && (namespacedList || len(args) != 4) {
```

Comment:
clearer with OR condition
```suggestion
			if len(args) != 4 || (namespacedList && len(args) != 3) {
```